### ISSUE TITLE
Webmaker Ideation

### ISSUE THREAD

**Issue by xmatthewx** [mozilla/webmaker-core#582]
Over the next month, we will examine the use of Webmaker, discuss the lessons we learned while building it, and revisit the roadmap alongside the Mozilla Learning strategic plan. 

Please comment freely below:
- outline topics to discuss
- link to interesting articles
- highlight products in the field
- ideate
  ... all in preparation for a sprint at the end of Sept

**Comment by xmatthewx** [comment]
#### Topic: Iteration

Frequent and well informed iteration is a key ingredient to success for many digital products.
- How can we test, learn, and iterate more frequently and more easily?
  - How can we spend more time with users? (watch: [Tom Chi](https://air.mozilla.org/july-brantina-on-prototyping-with-tom-chi/))
- How can we increase experimentation and reduce the corresponding risk?

**Comment by xmatthewx** [comment]
#### Topic: Social
- How does social foster participation online?
- How does it differ within learning spaces, answer forums, and social spaces?
- Learn about online communities: https://github.com/mozilla/webmaker-core/issues/579

**Comment by xmatthewx** [comment]
#### Topic: Product as intervention

How can our tools help level up people's engagement with the web?

Laura's research includes a framework for thinking about users relationship to the web, a spectrum of engagement: No Use, Limited Use, Leverage, Creation. (I would add Advocates & Architects to that list -- that's us and our fellows.)

Her research also identifies the blockers and requirements for a person to level-up. 
- How can we intervene around specific blockers and requirements? How can our tools directly or indirectly lead to higher-level engagement? 

[User journey plateaus](https://cloud.githubusercontent.com/assets/53411/9445535/ac53584e-4a5a-11e5-86b2-362a983dd74b.png)

**Comment by xmatthewx** [comment]
#### Topic: Code

Among other web literacies, code knowledge unlocks a person's ability to understand the web and shape its future. 
- Can code enhance Webmaker’s new platform for content discovery, making, and sharing? How? Why? What’s possible, probable, desirable?
- What does research show that works/fails in teaching code online?
- Who needs to learn it? Why?
- Can Webmaker provide mentor-like guidance around the most common errors?
- How is code useful/not useful for different people? Code curious, advanced makers, eager sharer. 
- Where is Webmaker situated relative to: blockly, scratch, code academy, tumblr (themes), wordpress (plugins)?

[Comparing Textual and Block Interfaces in a Novice Programming Environment](https://drive.google.com/a/mozillafoundation.org/file/d/0B_rbDAen9prkcGVnZFpwYm5TaHc/view)

**Comment by xmatthewx** [comment]
#### Topic: Live the mission

As the org strives to elevate mentors, teachers, advocates, leaders, how can we emulate that in our own practice? How can we make time spent as a mentor or advocate more integral to our monthly work routine?

**Comment by xmatthewx** [comment]
#### Topic: Offline to online

MoFo has a strong ground game with fellows, hives, Maker Party etc. How can Webmaker connect offline and online engagement? Are there opportunities at both ends of the engagement spectrum?

**Comment by xmatthewx** [comment]
Add thoughts, experience, links, topics, ideas, etcetera ... @flukeout @LauraReynal @simonwex @gvn @k88hudson @vazquez @Pomax @adamlofting @HPaulJohnson

**Comment by adamlofting** [comment]
This is already on the roadmap and in the existing thinking, but just to keep it on the radar:

Re product as intervention:
- What keeps people coming back to a product? (assuming that the longer our relationship with our users the better we can teach, advocate, inlfluence them)
  - Why should I come back on day 1, day 7, day 30, day 90? And what will trigger this behaviour?
- Is the long term user facing value of the product different from the chance to 'intervene'. E.g. Firefox is used repeatedly because of it's function as a browser, which then offers channels to engage on policy issues.

**Comment by lauradereynal** [comment]
:+1: to @adamlofting we had a great conversation this morning about which feature would drive user retention and how this would affect the roadmap. we should look at this together

**Comment by xmatthewx** [comment]
Definitely. What brings someone back to Webmaker beyond their initial need or desire? What specific features can be optimized to serve as hooks for return visits? (To clarify, I meant intervention by us to unblock users, not user intervention through participation in a campaign.)

Where can and should we intervene to help users reach the next plateau? 

<img width="795" alt="screenshot 2015-08-24 12 16 50" src="https://cloud.githubusercontent.com/assets/53411/9445679/79ffc7f0-4a5b-11e5-8745-86bebd07cb33.png">

**Comment by vazquez** [comment]
#### Design

With an understanding that well-designed web content has the capacity to communicate overwhelmingly well, how can we push towards teaching design principles to our community? What design principles are the most important to convey to others? What do we need do to allow for people to create naturally beautiful content? Should we teach design alongside code?

**Comment by xmatthewx** [comment]
Great addition @vazquez. I talked with @secretrobotron about finding ways to support users with interesting content to elevate their work even further. Would be great to find async and realtime ways to do this.

**Comment by xmatthewx** [comment]
#### Topic: Impact

Mozilla Learning [strategic planning](https://wiki.mozilla.org/Learning/impact_howto) strives to identify clear statements of impact to guide our roadmaps. What is Webmaker's impact? Drop ideas below.

**Mozilla Learning impact**: Many more people understand and care about the internet.

A few ideas for Webmaker:

 _1_ More people create and share on the open web (regardless of their device).
 _2_ More people understand and leverage the web’s power.
 _3_ More people leverage the web to meet their needs and desires.
 _4_ More people create and share on safe, people-first platforms.
 _5_ More people create and share content that positively impacts other people’s lives.
 _6_ More people create and share content for their local community.

**Comment by lauradereynal** [comment]
Regarding Impact : 

1) and 6) seems the closest to reality for me. 2-5 are too big for what Webmaker is doing I think.

**Comment by lauradereynal** [comment]
I <3 the design addition from @vazquez 

@xmatthewx Im getting lost in the number of topics. How about we focus on 1 at a time ? Or reduce the scope ?

**Comment by xmatthewx** [comment]
@LauraReynal - Definitely. I wanted a meta bug to help with planning. For let's dig into two:  
- social/community #579 - let's see who's interested, and then schedule a conversation 
- impact - we need to articulate how Webmaker connects with the Moz Learning strategy

**Comment by Timboliu** [comment]
In the current situation WebMaker focuses on creating content like pages, images, etc. Does WebMaker consider the possibility to make applications? I think this would fit great in learning about the web.

**Comment by xmatthewx** [comment]
@Timboliu thanks for the feedback. An earlier prototype of Webmaker explored applications, but we found the complexity didn't match well with the needs we found in our research.

**Comment by Timboliu** [comment]
@xmatthewx Thanks for your reply. Do you have more information about that research? I'm curious how this research was conducted.

**Comment by Timboliu** [comment]
I came across Paper today (https://vimeo.com/138268307). I think this app is really beautiful and maybe contains some functionality that could be developed in WebMaker? What I like is the paper-look-and-feel. I also like it when you open a project and you see all the papers of that project.

**Comment by Timboliu** [comment]
Regarding my previous question about creating applications. Maybe the creation of applications will supported by the Thimble project (https://thimble.mozilla.org)?
